io/disk/tiobench.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Rajashree Rajendran<rajashr7@linux.vnet.ibm.com>
+
+# Based on code by
+#   Author: Lucas Meneghel Rodrigues <lmr@redhat.com>
+#   Author: Xu Tian <xutian@redhat.com>
+#   Author: Jeff Moyer <jmoyer@redhat.com>
+#   Author: Dmitry Monakhov <dmonakhov@openvz.org>
+#   Copyright: 2008 - 2015 Red Hat, Inc.
+#   Copyright: 2013 Red Hat, Inc.
+#   Copyright: 2010 Red Hat, Inc.
+#   Copyright: 2012 <dmonakhov@openvz.org>
+#   https://github.com/autotest/autotest-client-tests/tree/master/tiobench
+
+"""
+This program runs a multi-threaded I/O benchmark test
+to measure file system performance.
+"""
+
+import os
+
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Tiobench(Test):
+    """
+    Avocado module for tiobench.
+    """
+    def setUp(self):
+        """
+        Build tiobench.
+        Source:
+        https://github.com/mkuoppal/tiobench.git
+        """
+        s_mngr = SoftwareManager()
+        if not s_mngr.check_installed("gcc") and not s_mngr.install("gcc"):
+            self.error('Gcc is needed for the test to be run')
+        locations = ["https://github.com/mkuoppal/tiobench/archive/master.zip"]
+        tarball = self.fetch_asset("tiobench.zip", locations=locations)
+        archive.extract(tarball, self.srcdir)
+        os.chdir(os.path.join(self.srcdir, "tiobench-master"))
+        build.make(".")
+
+    def test(self):
+        """
+        Test execution with necessary arguments.
+        :params target: The directory in which to test.
+                        Defaults to ., the current directory.
+        :params blocks: The blocksize in Bytes to use. Defaults to 4096.
+        :params threads: The number of concurrent test threads.
+        :params size: The total size in MBytes of the files may use together.
+        :params num_runs: This number specifies over how many runs
+                          each test should be averaged.
+        """
+        target = self.params.get('target', default='/var/tmp')
```

Comment:
Actually I'd even change this to:

```python
target = self.params.get('target') or self.workdir
```
To handle invalid input gracefully, what do you think? (eg. when one specifies `target: None` in the yaml file)